ansible-scylla-node/tasks/start_one_node.yml

```diff
@@ -1,10 +1,109 @@
 ---
+- name: Check if current node is already bootstrapped
+  shell: |
+    nodetool status | grep -o {{ item }}
+  ignore_errors: true
+  delegate_to: "{{ scylla_seeds[0] }}"
+  register: _bootstrapped
+  when: replication_aware_token_distributor is defined
+
+- name: Replication aware token distributor tasks
+  block:
+    - name: Get current token ring
+      block:
+        - name: Get tokens
+          shell: |
+            for ip in `nodetool status | grep -Eo '[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}'`; do
+                echo "$ip=`curl -s -X GET http://localhost:10000/storage_service/tokens/$ip | cut -d '[' -f2 | cut -d ']' -f1 | tr -d \\" | tr -d \ `"
+            done
```

Comment:
This looks terribly over complicated and messy.
Why couldn't you:
1) Calculate tokens for all nodes in the cluster (based on the inventory.ini) and define a per-host list that contains these tokens.
2) Leave the Role's code that bootstraps nodes as it does today except for:
  * If node is actually being bootstrapped (**) you:
    * Set it pre-calculated tokens using `initial_token` parameter
    * Bootstrap.
    * Remove `initial_token`.

(**) to identify which nodes need to be bootstrapped you can use a `nodetool ring` or the corresponding REST API (preferably) to fetch all known nodes and their current state:
 * If there is any node not in a "NORMAL" state - exit with an error.

This needs to be executed on any alive node: alive nodes can be checked using CQL port probing.

And all this needs to be done before the bootstrapping loop and NOT in `start_one_node.yml`.

`start_one_node.yml` needs to use already pre-calculated variables and simply act on them as described above.